sdk/storage/inc/shares/share_directory_client.hpp

```diff
@@ -0,0 +1,189 @@
+     * directory returned from the server.
+     */
+    Azure::Core::Response<SetDirectoryPropertiesResult> SetProperties(
+        FileShareSmbProperties smbProperties,
+        const SetDirectoryPropertiesOptions& options = SetDirectoryPropertiesOptions()) const;
+
+    /**
+     * @brief Sets the metadata of the directory.
+     * @param metadata User-defined metadata to be stored with the directory. Note that the string
+     *                 may only contain ASCII characters in the ISO-8859-1 character set.
+     * @param options Optional parameters to set this directory's metadata.
+     * @return Azure::Core::Response<SetDirectoryMetadataResult> containing the information of the
+     * directory returned from the server.
+     */
+    Azure::Core::Response<SetDirectoryMetadataResult> SetMetadata(
+        const std::map<std::string, std::string>& metadata,
+        const SetDirectoryMetadataOptions& options = SetDirectoryMetadataOptions()) const;
+
+    /**
+     * @brief List files and directories under the directory.
+     * @param options Optional parameters to list the files and directories under this directory.
+     * @return Azure::Core::Response<ListFilesAndDirectoriesSegmentedResult> containing the
+     * information of the operation, directory, share and the listed result.
+     */
+    Azure::Core::Response<ListFilesAndDirectoriesSegmentedResult> ListFilesAndDirectoriesSegmented(
+        const ListFilesAndDirectoriesSegmentedOptions& options
+        = ListFilesAndDirectoriesSegmentedOptions()) const;
+
+    /**
+     * @brief List open handles on the directory.
+     * @param options Optional parameters to list this directory's open handles.
+     * @return Azure::Core::Response<ListDirectoryHandlesSegmentedResult> containing the information
+     * of the operation and the open handles of this directory
+     */
+    Azure::Core::Response<ListDirectoryHandlesSegmentedResult> ListHandlesSegmented(
+        const ListDirectoryHandlesSegmentedOptions& options
+        = ListDirectoryHandlesSegmentedOptions()) const;
+
+    /**
+     * @brief Closes a handle or handles opened on a directory at the service.
+     * @param handleId The ID of the handle to be closed.
+     * @param options Optional parameters to close one of or all this directory's open handles.
+     * @return Azure::Core::Response<ForceCloseDirectoryHandlesResult> containing the information
+     * of the closed handles
+     * @remark This operation may return a marker showing that the operation can be continued.
+     */
+    Azure::Core::Response<ForceCloseDirectoryHandlesResult> ForceCloseHandles(
+        const std::string& handleId,
+        const ForceCloseDirectoryHandlesOptions& options
+        = ForceCloseDirectoryHandlesOptions()) const;
+
+  private:
+    UriBuilder m_shareDirectoryUri;
+    std::shared_ptr<Azure::Core::Http::HttpPipeline> m_pipeline;
```

Comment:
these two should be protected?

Reply:
Why? I don't think there will should child class for `DirectoryClient`.